.github/workflows/backport.yml

```
@@ -0,0 +1,83 @@
+name: backport
+
+on:
+  push:
+    branches:
+      - 'main'
+
+jobs:
+  backport-target-branch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Extract pr_number from commit message
+        run: |
+          latest_commit=$(git rev-parse HEAD)  # 현재 repository의 가장 최근 commit 해시 가져오기
+          echo "latest_commit=$latest_commit" >> $GITHUB_OUTPUT
+          commit_message=$(git show -s --format=%B $latest_commit)  # 최근 commit의 메시지 가져오기
+          echo "commit_message=$commit_message" >> $GITHUB_OUTPUT
+          pr_number=$(echo $commit_message | grep -oP "\(#\d+\)$" | grep -oP "\d+")  # commit 메시지에서 pr 번호 가져오기
+      - name: Check if this is a merged pr commit
+        run: |
+          if [[ -z "$pr_number" ]]; then
+            echo "This is not a merged pr commit."
+            exit 0
+          fi
+      - name: Get latest release
+        run: |
+          latest_release=$(cat VERSION | grep -oP "\d+\.\d+")
+          echo "latest_release=$latest_release" >> $GITHUB_OUTPUT
+      - name: Get target milestone
+        run: |
+          tartget_milestone=$(gh pr view $pr_number --json milestone.title --jq .milestone.title)
+      - uses: actions/checkout@v4
+      - name: Get milestone list
+        run: |
+          milestones=$(gh api /repos/:owner/:repo/milestones --jq '.[].title')
+      - name: Remove Backlog from milestone list
+        run: |
+          milestones=($milestones)
+          for i in "${!milestones[@]}"; do
+            if [[ "${milestones[$i]}" == "Backlog" ]]; then
+              unset 'milestones[$i]'
+            fi
+          done
+      - name: Sort milestone list
+        run: |
+          sort_milestones=($(printf "%s\n" "${milestones[@]}" | sort))
+      - name: Get target milestones
+        run: |
+          for i in "${!sort_milestones[@]}"; do
+            echo "$i: ${sort_milestones[$i]}"
+            if [[ "${sort_milestones[$i]}" == "$target_milestone" ]]; then
+              target_milestones=("${sort_milestones[@]:0:$((i+1))}")
+              break
+            fi
+          done
+          echo "matrix={\"milestone\":[$(printf "%s," "${target_milestones[@]}")]}" >> $GITHUB_OUTPUT
+
+  backport:
+    runs-on: ubuntu-latest
+    needs: backport-target-branch
+    strategy:
+      matrix: ${{ fromJson(needs.backport-target-branch.outputs.matrix) }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ matrix.milestone }}
+      - name: Cherry-pick
+        run: |
+          git cherry-pick ${{ needs.backport-target-branch.outputs.latest_commit }}
+          git commit --amend -m "${{ needs.backport-target-branch.outputs.commit_message }} Backported-from: main($latest_release) Backported-to: ${{ matrix.milestone }} Backport-of: ${{ needs.backport-target-branch.outputs.pr_number }}"
```

Suggestion:
```suggestion
          git commit --amend -m "${{ needs.backport-target-branch.outputs.commit_message }}" \
            --trailer "Backported-from=main($latest_release)" \
            --trailer "Backported-to: ${{ matrix.milestone }}" \
            --trailer "Backport-of: ${{ needs.backport-target-branch.outputs.pr_number }}"
```

---

I learned that trailers add text to commits.
In that case, it would be good to just use the trailer option without amend.
```suggestion
          git commit \
            --trailer "Backported-from=main($latest_release)" \
            --trailer "Backported-to=${{ matrix.milestone }}" \
            --trailer "Backport-of=${{ needs.backport-target-branch.outputs.pr_number }}"
```